### PR TITLE
Add Formal Languages and Compilers

### PR DESCRIPTION
**Pull request by patrickporto**
https://github.com/patrickporto/awesome-compilers-languages

Programming language design, domain specific languages and homebrew compilers

I confirm I've read and complied with the below requirements 🖖

### CODE REVIEW

**Comment by patrickporto** [comment]
unicorn

File: readme.md [hunk 1]
```diff
@@ -334,6 +334,7 @@
 - [Static Analysis & Code Quality](https://github.com/analysis-tools-dev/static-analysis#readme)
 - [Information Retrieval](https://github.com/harpribot/awesome-information-retrieval#readme) - Learn to develop your own search engine.
 - [Quantum Computing](https://github.com/desireevl/awesome-quantum-computing#readme) - Computing which utilizes quantum mechanics and qubits on quantum computers.
+- [Formal Languages & Compilers](https://github.com/patrickporto/awesome-compilers-languages#readme) - Programming language design, domain specific languages and homebrew compilers
```
Comment (Sea-n) [inline readme.md:337]: The description should end in a dot.

**Comment by probonopd** [comment]
Hi @patrickporto. As per [the guidelines](https://github.com/sindresorhus/awesome/blob/main/pull_request_template.md#requirements-for-your-awesome-list),

> Entries have a description, unless the title is descriptive enough by itself. It rarely is though.

Please add descriptions to the links.

**Review by JoseDeFreitas — CHANGES_REQUESTED** [review]
According to the [guidelines of the PRs and the list](https://github.com/sindresorhus/awesome/blob/main/pull_request_template.md):
> The default branch should be named main, not master.

The default branch in the repository is `master`, you must change it to `main`.

**Comment by sindresorhus** [comment]
I think the list is a good idea, but it needs more work and content before being something that can be added here.

I also don't see the point of linking to every part of a multi-part tutorial. Linking to the first lesson should be enough.

I'm going to close this for now, but happy to reconsider in the future when the list is more mature and has more content.

**Pull request closed without merging**